packages/block-library/src/navigation-link/edit.js

```diff
@@ -164,21 +165,23 @@ function NavigationLinkEdit( {
 						] }
 					/>
 					{ isLinkOpen && (
-						<LinkControl
-							className="wp-block-navigation-link__inline-link-input"
-							value={ link }
-							onChange={ ( {
-								title: newTitle = '',
-								url: newURL = '',
-								opensInNewTab: newOpensInNewTab,
-							} = {} ) => setAttributes( {
-								title: escape( newTitle ),
-								url: newURL,
-								label: label || escape( newTitle ),
-								opensInNewTab: newOpensInNewTab,
-							} ) }
-							onClose={ () => setIsLinkOpen( false ) }
-						/>
+						<Popover position="bottom center">
+							<LinkControl
+								className="wp-block-navigation-link__inline-link-input"
+								value={ link }
+								onChange={ ( {
+									title: newTitle = '',
+									url: newURL = '',
+									opensInNewTab: newOpensInNewTab,
+								} = {} ) => setAttributes( {
+									title: escape( newTitle ),
+									url: newURL,
+									label: label || escape( newTitle ),
+									opensInNewTab: newOpensInNewTab,
+								} ) }
+								onClose={ () => setIsLinkOpen( false ) }
```

Comment:
This should have been moved to the `Popover`. `LinkControl` no longer has an `onClose` prop with the changes from this pull request.

Fix at: #19885